deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Convolution3D.java

```diff
@@ -0,0 +1,193 @@
+package org.deeplearning4j.nn.conf.layers;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.deeplearning4j.nn.api.Layer;
+import org.deeplearning4j.nn.api.ParamInitializer;
+import org.deeplearning4j.nn.conf.ConvolutionMode;
+import org.deeplearning4j.nn.conf.InputPreProcessor;
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.conf.inputs.InputType;
+import org.deeplearning4j.nn.layers.convolution.Convolution3DLayer;
+import org.deeplearning4j.nn.params.Convolution3DParamInitializer;
+import org.deeplearning4j.optimize.api.IterationListener;
+import org.deeplearning4j.util.Convolution3DUtils;
+import org.deeplearning4j.util.ConvolutionUtils;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * 3D convolution layer configuration
+ *
+ * @author Max Pumperla
+ */
+@Data
+@NoArgsConstructor
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+public class Convolution3D extends ConvolutionLayer {
+
+    private ConvolutionMode mode = ConvolutionMode.Same;  // in libnd4j: 0 - same mode, 1 - valid mode
+    private boolean isNCDHW = true; // in libnd4j: 1 - NCDHW, 0 - NDHWC
```

Comment:
Would it be clearer to have an enumeration here instead of a boolean? Also a bit more flexible in case we ever have to change (or add another) format.